app/controllers/work_controller.rb

```diff
@@ -246,6 +246,12 @@ def update_featured_page
     redirect_back fallback_location: @work
   end
 
+  def document_sets_select
+    document_sets = current_user.document_sets.where(collection_id: params[:collection_id])
+
+    render partial: 'document_sets_select', locals: { document_sets: document_sets }
+  end
```

Comment:
The secret sauce. We have a route that, instead of rendering a full response, will render a partial instead.